ml.q

```diff
@@ -258,12 +258,20 @@ cgroup:{[df;X;C]value group imin f2nd[df X] C}
 / matri(X) to the nearest (C)entroid and then uses the (c)entroid (f)unction
 / to update the centroid location.
 lloyd:{[df;cf;X;C]cf X@\: cgroup[df;X;C]}
+/ use (r)esponsibility (f)unction David Mackay's Information Theory..(pg289)
+lloyds:{[df;cf;rf;X;C] cf[rf .ml.f2nd[df X] C;X]} /soft assignment 
 
 kmeans:lloyd[edist2;avg'']      / k-means
 kmedians:lloyd[mdist;med'']     / k-medians
 khmeans:lloyd[edist2;hmean'']   / k harmonic means
 skmeans:lloyd[cosdist;normalize (avg'')::] / spherical k-means
 
+kmeanss:lloyds[edist2;wavg\:/:;dax[=;min]] /k-means using loyd with rf
+/ kmeansoft v1 David Mackay (b)eta stiffness param 
+/ sigma or radius of cluster is 1%sqrt b
+smin:prb exp@* /soft min
```

Comment:
ah, now i see the pattern.  this is in-fact the ``.ml.softmax function``.  instead of adding this new function, perhaps just just add ``softmax neg[b]*`` into ``kmeanssmin`` 

how about renaming it to ``kmeanssmax``?

Reply:
Agreed, thanks for reviewing and of course the code base already had all these primitives, but somehow it's hard to recall them if you didn't write the code yourself.  :)
softmin is negative beta of softmax, is that confusing?

Reply:
also I found sometimes  a wavg using a boolean vector is faster than selecting the data and taking the average (after accounting for type conversions)
i:10000000;af:9h$a:i?0b;b:i?9h$i;
\ts af wavg b
\ts avg b where a